File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryConsts.java
/*
 * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.destination.bigquery;

public class BigQueryConsts {

  public static final int MiB = 1024 * 1024;
  public static final String CONFIG_DATASET_ID = "dataset_id";
  public static final String CONFIG_PROJECT_ID = "project_id";
  public static final String CONFIG_DATASET_LOCATION = "dataset_location";
  public static final String CONFIG_CREDS = "credentials_json";
  public static final String BIG_QUERY_CLIENT_CHUNK_SIZE = "big_query_client_buffer_size_mb";

  public static final String LOADING_METHOD = "loading_method";
  public static final String METHOD = "method";
  public static final String GCS_STAGING = "GCS Staging";
  public static final String GCS_BUCKET_NAME = "gcs_bucket_name";
  public static final String GCS_BUCKET_PATH = "gcs_bucket_path";
  public static final String GCS_BUCKET_REGION = "gcs_bucket_region";
  public static final String CREDENTIAL = "credential";
  public static final String FORMAT = "format";
  public static final String KEEP_GCS_FILES = "keep_files_in_gcs-bucket";
  public static final String KEEP_GCS_FILES_VAL = "Keep all tmp files in GCS";

  public static final String NAMESPACE_PREFIX = "n";

  // tests
  public static final String BIGQUERY_BASIC_CONFIG = "basic_bigquery_config";
  public static final String GCS_CONFIG = "gcs_config";

  public static final String CREDENTIAL_TYPE = "credential_type";
  public static final String HMAC_KEY_ACCESS_ID = "hmac_key_access_id";
  public static final String HMAC_KEY_ACCESS_SECRET = "hmac_key_secret";

}

File: airbyte-integrations/connectors/destination-bigquery/src/main/kotlin/io/airbyte/integrations/destination/bigquery/BigQueryConsts.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.integrations.destination.bigquery

object BigQueryConsts {
    const val MiB: Int = 1024 * 1024
    const val CONFIG_DATASET_ID: String = "dataset_id"
    const val CONFIG_PROJECT_ID: String = "project_id"
    const val CONFIG_DATASET_LOCATION: String = "dataset_location"
    const val CONFIG_CREDS: String = "credentials_json"
    const val BIG_QUERY_CLIENT_CHUNK_SIZE: String = "big_query_client_buffer_size_mb"

    const val LOADING_METHOD: String = "loading_method"
    const val METHOD: String = "method"
    const val GCS_STAGING: String = "GCS Staging"
    const val GCS_BUCKET_NAME: String = "gcs_bucket_name"
    const val GCS_BUCKET_PATH: String = "gcs_bucket_path"
    const val GCS_BUCKET_REGION: String = "gcs_bucket_region"
    const val CREDENTIAL: String = "credential"
    const val FORMAT: String = "format"
    const val KEEP_GCS_FILES: String = "keep_files_in_gcs-bucket"
    const val KEEP_GCS_FILES_VAL: String = "Keep all tmp files in GCS"
    const val DISABLE_TYPE_DEDUPE: String = "disable_type_dedupe"
    const val RAW_DATA_DATASET = "raw_data_dataset"
    const val NAMESPACE_PREFIX: String = "n"
}
